AWSAppSyncClient/Internal/AWSPerformMutationOperation.swift

```diff
@@ -29,35 +36,61 @@ final class AWSPerformMutationOperation<Mutation: GraphQLMutation>: Asynchronous
         self.mutation = mutation
         self.mutationConflictHandler = mutationConflictHandler
         self.mutationResultHandler = mutationResultHandler
+        super.init()
+        determineMutationStep()
     }
 
     deinit {
         AppSyncLog.verbose("\(identifier ?? "(no identifier)"): deinit")
     }
+    
+    private func determineMutationStep() {
```

Comment:
If we do rename "next step" to "state", rename this to `resolveInitialMutationState` or similar